docs/semgrep-appsec-platform/semgrep-api.md

```diff
@@ -0,0 +1,25 @@
+---
+slug: semgrep-api
```

Comment:
An entirely new doc? Por que?

Reply:
AHHH we changed the name of the folder. Hmm. I was thinking of moving most deployment-oriented features out to /deployment, thereby future-proofing them from rebrands. The /semgrep-appsec-platform will likely survive, but with fewer docs. 

I'll apply `mv` changes on this PR, is that ok?

Reply:
Yes, please go ahead!